File: consumer-web/src/foam.ts
import * as foam from '@foam-ai/node-opentelemetry';
import { FOAM_API_KEY, IS_PRODUCTION } from './config/keys';
import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express';
import type { InstrumentationBase } from '@opentelemetry/instrumentation';

foam.init({
  serviceName: 'consumer-web',
  isProduction: IS_PRODUCTION,
  apiKey: `Bearer ${FOAM_API_KEY}`,
  additionalInstrumentations: [
    new ExpressInstrumentation({}) as unknown as InstrumentationBase
  ],
  captureConsoleErrors: true,
  captureUnhandledRejections: true
});

File: online-storage-proxy/src/foam.py
import foam
import asyncio
from config import FOAM_API_KEY, IS_PRODUCTION

# Initialize Foam observability
foam.init(
    service_name="online-storage-proxy",
    api_key=FOAM_API_KEY,
    is_production=IS_PRODUCTION,
    capture_sql=True,
    capture_connection_pools=True
)

if __name__ == "__main__":
    asyncio.run(main())